core/src/main/java/org/infinispan/interceptors/distribution/ScatteredDistributionInterceptor.java

```diff
@@ -221,10 +231,8 @@ private Object handleWriteCommand(InvocationContext ctx, DataWriteCommand comman
             } else {
                contextEntry = cacheEntry;
             }
-            return invokeNextThenApply(ctx, command, (rCtx, rCommand, rv) -> {
-               commitSingleEntryIfNewer(contextEntry, rCtx, rCommand);
-               return null;
-            });
+            return invokeNextThenApply(ctx, command, (rCtx, rCommand, rv) ->
+                  stageOrNullSyncValue(commitSingleEntryIfNewer(contextEntry, rCtx, rCommand)));
```

Comment:
The main reason I wouldn't always do that is then you can't tell easily if the `InvocationStage` is complete or not for sync invocations. Now I guess one could argue you don't need to know, if you cache all the lambdas (which requires them to be non capturing). Unfortunately a lot capture the return value from the interceptor sadly. https://github.com/infinispan/infinispan/pull/6401/files/23f09962b642dc97340613d25a675f06c42bd425#diff-e82d8c3460dda205aac1ddef3152328eR221

But I see you also mention something about inlining there. I don't see how it could get rid of the lambda in that it is capturing and it may have to execute it later if the stage isn't complete yet.